src/test/java/org/ambraproject/rhino/service/impl/ArticleListCrudServiceImplTest.java

```diff
@@ -0,0 +1,404 @@
+      assertThat(exception).hasMessageThat().isEqualTo("Journal not found: " + identity.getJournalKey());
+    }
+  }
+
+  @Test
+  public void testCreate_FailForExistingArticleList() throws Exception {
+    ArticleListIdentity identity = createStubArticleListIdentity();
+    String displayName = ARTICLE_LIST_TITLE;
+    Set<ArticleIdentifier> articleIds = new HashSet<>();
+
+    when(mockHibernateTemplate.execute(any())).thenReturn(1L);
+
+    try {
+      mockArticleListCrudService.create(identity, displayName, articleIds);
+      fail("Expecting exception, but nothing was thrown.");
+    } catch (Exception exception) {
+      assertThat(exception).isInstanceOf(RestClientException.class);
+      assertThat(exception).hasMessageThat().isEqualTo("List already exists: " + identity);
+    }
+  }
+
+  @Test
+  public void testUpdate_SucceedForDisplayName() {
+    ArticleListIdentity identity = createStubArticleListIdentity();
+    ArticleListView articleListView = createStubArticleList();
+
+    when(mockHibernateTemplate.execute(any())).thenReturn(
+        new ArrayList(Collections.singletonList(new Object[] {
+            "valid-journal", articleListView.getArticleList()
+        })));
+
+    String newDisplayName = "article list 2";
+    ArticleListView newArticleListView = mockArticleListCrudService.update(identity, Optional.ofNullable(newDisplayName), Optional.empty());
+    assertThat(newArticleListView.getIdentity()).isEqualTo(identity);
+    assertThat(newArticleListView.getArticleList().getDisplayName()).isEqualTo(newDisplayName);
+    assertThat(newArticleListView.getArticleList().getArticles()).isEmpty();
+
+  }
+
+  @Test
+  public void testUpdate_SucceedForAddArticlesList() {
+    ArticleListIdentity identity = createStubArticleListIdentity();
+    ArticleListView articleListView = createStubArticleList();
+
+    Article article = createStubArticle();
+    Set<String> articleKeys = new HashSet<String>();
+    articleKeys.add(article.getDoi());
+
+    when(mockHibernateTemplate.execute(any())).thenReturn(
+        new ArrayList(Collections.singletonList(new Object[] {
+            "valid-journal", articleListView.getArticleList()
+        })));
+
+    when(mockHibernateTemplate.findByNamedParam("from Article where doi in :articleKeys",
+        "articleKeys", articleKeys)).thenReturn(
+        new ArrayList(Collections.singletonList(article))
+    );
+
+    Set<ArticleIdentifier> newArticleIds = new HashSet<>();
+    newArticleIds.add(ArticleIdentifier.create(article.getDoi()));
+
+    ArticleListView newArticleListView = mockArticleListCrudService.update(identity, Optional.empty(), Optional.ofNullable(newArticleIds));
+    assertThat(newArticleListView.getIdentity()).isEqualTo(identity);
+    assertThat(newArticleListView.getArticleList().getArticles().size()).isEqualTo(newArticleIds.size());
+  }
+
+  @Test
+  public void testUpdate_SucceedForRemoveArticlesList() {
+    ArticleListIdentity identity = createStubArticleListIdentity();
```

Comment:
This line is common in a lot of tests, you should consider moving it to a common init() method.

Reply:
If it is in init() (or someother) method, then identity and articleListView will need to be defined as member variables. So will make it harder to read within the test function.I would like to leave it as it is.